app/modules/code_provider/github/github_service.py

```
@@ -422,13 +430,52 @@ async def get_combined_user_repos(self, user_id: str):
+                    branches = [
+                        branch.name
+                        for branch in local_repo.heads
+                        if branch.name != default_branch
+                    ]
+
+                    return {"branches": [default_branch] + branches}
+                except git.InvalidGitRepositoryError:
+                    raise HTTPException(
+                        status_code=404,
+                        detail=f"Not a valid git repository: {repo_name}",
+                    )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)